src/schema.rs

```diff
@@ -46,7 +46,7 @@ impl Schema {
 
     /// Returns the requested type, if present in this schema;
     /// otherwise returns None.
-    fn get_type<A: AsRef<str>>(&self, name: A) -> Option<&Type> {
+    fn get_type<A: AsRef<str>>(&self, name: A) -> Option<&TypeDefinition> {
```

Comment:
Is there a good reason to hand out a `&TypeDefinition` instead of a `TypeRef`? (Same question for all of the below method signatures in `Schema` too.)